### PR TITLE
WF-507:Dogfooding fixes

### PR DESCRIPTION
**Pull request by JonasImprobable**
- Moved the snapshot file from `./shared`  to `./SpatialOS/snapshots`.
- Added a known issues section to the readme.
- Various drive-by improvements to our docs.

cc @anne-edwards for reviewing changes to the readme.

### CODE REVIEW

File: docs/reference/client-worker-configuration.md [hunk 1]
```diff
@@ -14,36 +14,40 @@ The file may be called by any name but should have the following structure:
...
+    }
   }
 }
 ```
 
 | Field | Required/Optional | Description | 
 | :------------- | :------------- | :------- |
-| `workerType` | Required | Used to identify this worker type elsewhere in your project. |
-|`layer`| Optional | The name of the simulation layer of this worker, used to determine the load balancing strategy. For more information, see [introduction to layers](https://docs.improbable.io/reference/latest/shared/worker-configuration/layers#introduction-to-layers).|
+| `workerType` | Required | Used to identify this worker type elsewhere in your project. <br><br> **Note:** A `workerType` must be unique project-wide between server and client workers. The `workerType` specified in this field must also match the `workerType` that your worker reports in its `ConnectionParameters` when connecting to a deployment. For more infromation, see the [API reference](https://docs.improbable.io/reference/latest/csharpsdk/api-reference#improbable-worker-connectionparameters-class). |
```
Comment (ioanabalas) [inline docs/reference/client-worker-configuration.md]: infromation -> information

File: docs/reference/server-worker-configuration.md [hunk 2]
```diff
@@ -32,14 +41,14 @@ The file may be called by any name but should have the following structure:
 
 | Field | Required/Optional | Description | 
 | :------------- | :------------- | :------- |
-| `workerType` | Required | Used to identify this worker type elsewhere in your project. |
-|`layer`| Optional | The name of the simulation layer of this worker, used to determine the load balancing strategy. For more information, see [introduction to layers](https://docs.improbable.io/reference/latest/shared/worker-configuration/layers#introduction-to-layers).|
+| `workerType` | Required | Used to identify this worker type elsewhere in your project. <br><br> **Note:** A `workerType` must be unique project-wide between server and client workers. The `workerType` specified in this field must also match the `workerType` that your worker reports in its `ConnectionParameters` when connecting to a deployment. For more infromation, see the [API reference](https://docs.improbable.io/reference/latest/csharpsdk/api-reference#improbable-worker-connectionparameters-class). |
```
Comment (ioanabalas) [inline docs/reference/server-worker-configuration.md]: infromation -> information

File: HelloWorker/hello_worker.json [hunk 3]
```diff
@@ -24,6 +24,15 @@
       }
     },
     "localDeployment": {
+      "windows": {
```
Comment (frozenL) [inline HelloWorker/hello_worker.json]: why do we change orders?

File: README.md [hunk 4]
```diff
@@ -67,6 +67,16 @@ This connects a client that pings the `HelloWorker` and `DiceWorker` every few s
 ## Logs location
 The logs are stored in `logs` subdirectory of the location where your `spatialos.json` file is. For example, if my `spatialos.json` file is in `/workspace/project/SpatialOS`, the SpatialOS logs will go to `/workspace/project/SpatialOS/logs`.
 
+## Known Issues
+* `spatial local worker launch` is not supported.
```
Comment (achannn) [inline README.md]: `spatial local worker launch` is not supported. You will need to launch workers manually through the steps indicated in (link to the part of the doc above)

?

File: README.md [hunk 5]
```diff
@@ -67,6 +67,16 @@ This connects a client that pings the `HelloWorker` and `DiceWorker` every few s
...
+* `spatial project history snapshot convert` is not supported.
+* For [client workers](docs/reference/client-worker-configuration.md), it is currently not possible to specify the base command and command line arguments, with which a worker is started.
```
Comment (achannn) [inline README.md]: This seems related to point one. Maybe say something like how you just simply need to run the base command and command line arguments yourself instead of using a json?

File: README.md [hunk 6]
```diff
@@ -67,6 +67,16 @@ This connects a client that pings the `HelloWorker` and `DiceWorker` every few s
...
+* [Client workers](docs/reference/client-worker-configuration.md) that are not built using Unity or Unreal, while supported by the flexible project structure in general, are not supported by our [Launcher](https://docs.improbable.io/reference/latest/shared/operate/launcher#the-launcher) yet. As a result, you will not be able to start the client worker provided in this repository from the Launcher, if you deploy this project in the cloud.
```
Comment (achannn) [inline README.md]: Nit: This point is a bit wordy, I think we can remove the `while supported by the flexible project structure in general,` part.

File: docs/reference/client-worker-configuration.md [hunk 7]
```diff
@@ -14,36 +14,40 @@ The file may be called by any name but should have the following structure:
...
+  ],
   "componentDelivery": {
     "default": "RELIABLE_ORDERED",
     "checkoutAllInitially": true
   },
   "componentSettings": {
     ...
   },
-  "permissions": {
+  "permissions": [
```
Comment (achannn) [inline docs/reference/client-worker-configuration.md]: I don't think the permissions field is supposed to be an array

File: docs/reference/server-worker-configuration.md [hunk 8]
```diff
@@ -14,10 +14,19 @@ The file may be called by any name but should have the following structure:
...
+    ...
+  ],
   "componentDelivery": {
     "default": "RELIABLE_ORDERED",
     "checkoutAllInitially": true
   },
+  "componentSettings": {
+    ...
+  },
+  "permissions": [
```
Comment (achannn) [inline docs/reference/server-worker-configuration.md]: Same as client, don't think permissions is an array

File: docs/reference/client-worker-configuration.md (hunk 7 above)
Comment (JonasImprobable) [reply docs/reference/client-worker-configuration.md]: This is the format specified in our current docs: https://docs.improbable.io/reference/13.5/shared/worker-configuration/permissions#worker-permissions

File: docs/reference/server-worker-configuration.md (hunk 8 above)
Comment (JonasImprobable) [reply docs/reference/server-worker-configuration.md]: See comment above.

File: HelloWorker/hello_worker.json (hunk 3 above)
Comment (improbablepiotr) [reply HelloWorker/hello_worker.json]: I guess 90% of users are on Windows.

Comment (JonasImprobable) [reply HelloWorker/hello_worker.json]: The order change is here for consistency with our other docs including https://docs.improbable.io/reference/13.5/shared/worker-configuration/launch-configuration#worker-launch-configuration and https://github.com/spatialos/FlexibleProjectExample/blob/master/docs/reference/client-worker-configuration.md

File: README.md (hunk 4 above)
Comment (JonasImprobable) [reply README.md]: Re: How do we start workers in the future: Yes, users will need to manually start them.
Re: Should this be a known issue: Moved this note into a FAQ which will later become part of our migration guide.

File: README.md [hunk 9]
```diff
@@ -67,6 +67,16 @@ This connects a client that pings the `HelloWorker` and `DiceWorker` every few s
 ## Logs location
 The logs are stored in `logs` subdirectory of the location where your `spatialos.json` file is. For example, if my `spatialos.json` file is in `/workspace/project/SpatialOS`, the SpatialOS logs will go to `/workspace/project/SpatialOS/logs`.
 
+## Known Issues
+* `spatial local worker launch` is not supported.
+* `spatial project history snapshot convert` is not supported.
```
[the remark this answers is not in the document]
Comment (JonasImprobable) [reply README.md]: Moved to the FAQ page.

File: README.md (hunk 5 above)
Comment (JonasImprobable) [reply README.md]: Removed this point altogether.

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (JonasImprobable) [reply README.md]: Shortened sentence.

File: docs/reference/client-worker-configuration.md (hunk 7 above)
Comment (JonasImprobable) [reply docs/reference/client-worker-configuration.md]: Fixed. :)

File: docs/reference/client-worker-configuration.md (hunk 1 above)
Comment (JonasImprobable) [reply docs/reference/client-worker-configuration.md]: Fixed. :)

File: docs/reference/server-worker-configuration.md (hunk 8 above)
Comment (JonasImprobable) [reply docs/reference/server-worker-configuration.md]: Fixed.

File: docs/reference/server-worker-configuration.md (hunk 2 above)
Comment (JonasImprobable) [reply docs/reference/server-worker-configuration.md]: Fixed.

File: docs/migration-guide/faq.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# FAQ
+
+**Is `spatial local worker launch` supported?**<br/>
+`spatial local worker launch` is not supported and will not be supported going forward. Please start and connect your worker to a deployment manually. For more information, please refer to our [API documentation](https://docs.improbable.io/reference/latest/csharpsdk/api-reference#improbable-worker-connection-class).
```
Comment (achannn) [inline docs/migration-guide/faq.md]: The reference to the docs is a bit misleading, since even with `spatial local worker launch`, the worker would have had to use the same thing. The only thing spatial local worker launch did was run the command with args specified in the worker configs.

`You will need to start your worker manually as shown here [insert link to the part of the FlexibleProjectExample README where we show the commands to start a worker]` ?

File: docs/migration-guide/faq.md [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+
+**Is `spatial project history snapshot convert` supported?**<br/>
+`spatial project history snapshot convert` is not supported at the moment. We are planning to provide the functionality in the future.
```
Comment (achannn) [inline docs/migration-guide/faq.md]: ~the~ this

Comment (JonasImprobable) [reply docs/migration-guide/faq.md]: Applied your suggestion. :)

File: docs/migration-guide/faq.md (hunk 10 above)
Comment (JonasImprobable) [reply docs/migration-guide/faq.md]: Fixed.

File: README.md [hunk 12]
```diff
@@ -67,8 +67,10 @@ This connects a client that pings the `HelloWorker` and `DiceWorker` every few s
 ## Logs location
 The logs are stored in `logs` subdirectory of the location where your `spatialos.json` file is. For example, if my `spatialos.json` file is in `/workspace/project/SpatialOS`, the SpatialOS logs will go to `/workspace/project/SpatialOS/logs`.
 
-## Reference documentation
+## Known Issues
+* Our [Launcher](https://docs.improbable.io/reference/latest/shared/operate/launcher#the-launcher) currently only supports Unity and Unreal based [client workers](docs/reference/client-worker-configuration.md). As a result, you will not be able to start the client worker provided in this repository from the Launcher, if you deploy this project in the cloud.
```
Comment (anne-edwards) [inline README.md]:
```suggestion
* Our [Launcher](https://docs.improbable.io/reference/latest/shared/operate/launcher#the-launcher) currently only supports Unity and Unreal based [client-workers](docs/reference/client-worker-configuration.md). As a result, if you deploy this project to the cloud, you can't start the provided client-worker from the Launcher.
```

File: README.md [hunk 13]
```diff
@@ -78,3 +80,9 @@ The logs are stored in `logs` subdirectory of the location where your `spatialos
 [Server worker configuration](docs/reference/server-worker-configuration.md)
 
 [Client worker configuration](docs/reference/client-worker-configuration.md)
+
+## Migration Guide
+[FAQ](docs/migration-guide/faq.md)
```
Comment (anne-edwards) [inline README.md]: Are we planning to add a migration guide to replace this at some point?

File: docs/changelog.md [hunk 14]
```diff
@@ -0,0 +1,10 @@
+# Changelog
```
Comment (anne-edwards) [inline docs/changelog.md]: Should we do this the other way round instead, i.e. latest date at the top (like we do for release notes)? I'm not as familiar with change logs in general but it would make sense to me to switch the order.

File: docs/changelog.md [hunk 15]
```diff
@@ -0,0 +1,10 @@
...
+
+* **21 Jan 2019:** Added an upload configuration to the client worker.
```
Comment (anne-edwards) [inline docs/changelog.md]:
```suggestion
* **21 Jan 2019:** Added an upload configuration to the client-worker.
```

File: docs/changelog.md [hunk 16]
```diff
@@ -0,0 +1,10 @@
...
+* **22 Jan 2019:** Upgraded worker SDK version to `13.5.1`.
+* **30 Jan 2019:** 
+  * Moved the snapshot file from `./shared`  to `./SpatialOS/snapshots`.
+  * Added this changelog.
+  * Added a known issues section to the [readme](../README.md).
```
Comment (anne-edwards) [inline docs/changelog.md]: Could link directly to the known issues section instead of the readme?

File: docs/migration-guide/faq.md [hunk 17]
```diff
@@ -0,0 +1,7 @@
+# FAQ
+
+**Is `spatial local worker launch` supported?**<br/>
```
Comment (anne-edwards) [inline docs/migration-guide/faq.md]:
```suggestion
**Do you support `spatial local worker launch`?**<br/>
```
Passive -> active. Let me know if it's OK to say "you" here - hopefully it should be fairly clear what it means from the context.

File: docs/migration-guide/faq.md [hunk 18]
```diff
@@ -0,0 +1,7 @@
...
+`spatial local worker launch` is not supported and will not be supported going forward. Please start and connect your worker to a deployment manually as shown in the [readme](../../README.md#Running-the-project).
+
+**Is `spatial project history snapshot convert` supported?**<br/>
```
Comment (anne-edwards) [inline docs/migration-guide/faq.md]:
```suggestion
**Do you support `spatial project history snapshot convert`?**<br/>
```
As above.

File: docs/migration-guide/faq.md [hunk 19]
```diff
@@ -0,0 +1,7 @@
+# FAQ
+
+**Is `spatial local worker launch` supported?**<br/>
+`spatial local worker launch` is not supported and will not be supported going forward. Please start and connect your worker to a deployment manually as shown in the [readme](../../README.md#Running-the-project).
```
Comment (anne-edwards) [inline docs/migration-guide/faq.md]:
```suggestion
We don't support `spatial local worker launch`and won't support it going forward. Please start and connect your worker to a deployment manually as shown in the [readme](../../README.md#Running-the-project).
```

File: docs/migration-guide/faq.md [hunk 20]
```diff
@@ -0,0 +1,7 @@
...
+
+**Is `spatial project history snapshot convert` supported?**<br/>
+`spatial project history snapshot convert` is not supported at the moment. We are planning to provide this functionality in the future.
```
Comment (anne-edwards) [inline docs/migration-guide/faq.md]:
```suggestion
We don't support `spatial project history snapshot convert` at the moment. We are planning to provide this functionality in the future.
```

File: docs/reference/client-worker-configuration.md [hunk 21]
```diff
@@ -31,19 +31,23 @@ The file may be called by any name but should have the following structure:
...
+    }
   }
 }
 ```
 
 | Field | Required/Optional | Description | 
 | :------------- | :------------- | :------- |
-| `workerType` | Required | Used to identify this worker type elsewhere in your project. |
-|`layer`| Optional | The name of the simulation layer of this worker, used to determine the load balancing strategy. For more information, see [introduction to layers](https://docs.improbable.io/reference/latest/shared/worker-configuration/layers#introduction-to-layers).|
+| `workerType` | Required | Used to identify this worker type elsewhere in your project. <br><br> **Note:** A `workerType` must be unique project-wide between server and client workers. The `workerType` specified in this field must also match the `workerType` that your worker reports in its `ConnectionParameters` when connecting to a deployment. For more information, see the [API reference](https://docs.improbable.io/reference/latest/csharpsdk/api-reference#improbable-worker-connectionparameters-class). |
```
Comment (anne-edwards) [inline docs/reference/client-worker-configuration.md]: I'm not sure what "unique project-wide between server and client workers" means - is it that within a project, you can only use a particular `workerType` once (so if you use it for a client-worker, you can't use it for a server-worker)? If so I'd maybe leave out "between server and client workers" and just stick with "unique project-wide". 

I've also changed "that your worker reports" to that your worker instance reports" as I _think_ that is more accurate?

```suggestion
| `workerType` | Required | Used to identify this worker type elsewhere in your project. <br><br> **Note:** A `workerType` must be unique project-wide between server and client workers. The `workerType` specified in this field must also match the `workerType` that your worker instance reports in its `ConnectionParameters` when connecting to a deployment. For more information, see the [API reference](https://docs.improbable.io/reference/latest/csharpsdk/api-reference#improbable-worker-connectionparameters-class). |
```

File: docs/reference/client-worker-configuration.md [hunk 22]
```diff
@@ -31,19 +31,23 @@ The file may be called by any name but should have the following structure:
...
+|`layer`| Optional | The name of the simulation layer of this worker, used to determine the load balancing strategy. For more information, see [introduction to layers](https://docs.improbable.io/reference/latest/shared/worker-configuration/layers#introduction-to-layers). |
```
Comment (anne-edwards) [inline docs/reference/client-worker-configuration.md]:
```suggestion
|`layer`| Optional | The name of the simulation layer of this worker type, used to determine the load balancing strategy. For more information, see [Layers](https://docs.improbable.io/reference/latest/shared/worker-configuration/Layers). |
```

Updated the link because I changed this yesterday with the release of Kraken :)

File: docs/reference/client-worker-configuration.md [hunk 23]
```diff
@@ -31,19 +31,23 @@ The file may be called by any name but should have the following structure:
...
\ No newline at end of file
+| `uploadConfiguration` | Optional | Specifies the client worker's build directory for each platform. If provided, the `spatial alpha cloud upload` CLI command will upload the given client worker enabling it to be used with the [Launcher](https://docs.improbable.io/reference/latest/shared/operate/launcher#the-launcher). <br><br> It contains three optional fields (`windows`, `macos`, `linux`) which each contain one required `localBuildDirectory` string field. The `localBuildDirectory` field specifies the directory containing the worker binary to be executed. <br><br> **Note:** Your built out worker must be self-contained to the `localBuildDirectory` and all files that your worker might depend on while running must be present in this folder. Files outside of this folder will not be uploaded to the cloud as part of your project assembly. |
```
Comment (anne-edwards) [inline docs/reference/client-worker-configuration.md]:
```suggestion
| `uploadConfiguration` | Optional | Specifies the client worker's build directory for each platform. If provided, the `spatial alpha cloud upload` CLI command uploads the given client worker, enabling you to use it with the [Launcher](https://docs.improbable.io/reference/latest/shared/operate/launcher#the-launcher). <br><br> It contains three optional fields (`windows`, `macos`, `linux`) which each contain one required `localBuildDirectory` string field. The `localBuildDirectory` field specifies the directory containing the worker binary to be executed. <br><br> **Note:** Your built-out worker must be self-contained in the `localBuildDirectory` and all files that your worker might depend on while running must be in this folder. Files outside this folder will not be uploaded to the cloud as part of your project assembly. |
```

File: docs/reference/project-configuration.md [hunk 24]
```diff
@@ -24,6 +24,6 @@ The file must be called `spatialos.json` and must have the following structure:
...
 | `projectName`  | Required  | The name of the project assigned to you by Improbable. It's used in the Console and for deployments. It should be between 3 and 32 characters long and contain letters, numbers and underscores. |
-| `schemaDescriptor` | Required | A relative (to the location of this file) or absolute path to the SpatialOS schema descriptor file. This is a binary file obtained by using the SpatialOS schema compiler on all of the project's `.schema` files. See the example in the `build.sh` to see how this file can be generated. <br> <br> This file is required in order for the SpatialOS Runtime to understand the schema used by all the workers. |
+| `schemaDescriptor` | Required | A relative (to the location of this file) or absolute path to the SpatialOS schema descriptor file. This is a binary file obtained by using the SpatialOS schema compiler on all of the project's `.schema` files. See the example in the `build.sh` to see how this file can be generated. <br><br> This file is required in order for the SpatialOS Runtime to understand the schema used by all the workers. |
```
Comment (anne-edwards) [inline docs/reference/project-configuration.md]:
```suggestion
| `schemaDescriptor` | Required | A relative (to the location of this file) or absolute path to the SpatialOS schema descriptor file. This is a binary file obtained by using the SpatialOS schema compiler on all of the project's `.schema` files. See the example in the `build.sh` of how to generate this file. <br><br> This file is required in order for the SpatialOS Runtime to understand the schema used by all the workers. |
```

Minor change - do change it back if what I've done makes no sense.

File: docs/reference/server-worker-configuration.md [hunk 25]
```diff
@@ -32,14 +41,14 @@ The file may be called by any name but should have the following structure:
 
 | Field | Required/Optional | Description | 
 | :------------- | :------------- | :------- |
-| `workerType` | Required | Used to identify this worker type elsewhere in your project. |
-|`layer`| Optional | The name of the simulation layer of this worker, used to determine the load balancing strategy. For more information, see [introduction to layers](https://docs.improbable.io/reference/latest/shared/worker-configuration/layers#introduction-to-layers).|
+| `workerType` | Required | Used to identify this worker type elsewhere in your project. <br><br> **Note:** A `workerType` must be unique project-wide between server and client workers. The `workerType` specified in this field must also match the `workerType` that your worker reports in its `ConnectionParameters` when connecting to a deployment. For more information, see the [API reference](https://docs.improbable.io/reference/latest/csharpsdk/api-reference#improbable-worker-connectionparameters-class). |
```
Comment (anne-edwards) [inline docs/reference/server-worker-configuration.md]: Same comment as earlier about removing "between server and client workers" (if I've understood this correctly)

```suggestion
| `workerType` | Required | Used to identify this worker type elsewhere in your project. <br><br> **Note:** A `workerType` must be unique project-wide between server and client workers. The `workerType` specified in this field must also match the `workerType` that your worker instance reports in its `ConnectionParameters` when connecting to a deployment. For more information, see the [API reference](https://docs.improbable.io/reference/latest/csharpsdk/api-reference#improbable-worker-connectionparameters-class). |
```

File: docs/reference/server-worker-configuration.md [hunk 26]
```diff
@@ -32,14 +41,14 @@ The file may be called by any name but should have the following structure:
...
+|`layer`| Optional | The name of the simulation layer of this worker, used to determine the load balancing strategy. For more information, see [introduction to layers](https://docs.improbable.io/reference/latest/shared/worker-configuration/layers#introduction-to-layers). |
```
Comment (anne-edwards) [inline docs/reference/server-worker-configuration.md]:
```suggestion
|`layer`| Optional | The name of the simulation layer of this worker type, used to determine the load balancing strategy. For more information, see [Layers](https://docs.improbable.io/reference/latest/shared/worker-configuration/layers). |
```

File: docs/reference/server-worker-configuration.md [hunk 27]
```diff
@@ -32,14 +41,14 @@ The file may be called by any name but should have the following structure:
...
 | `permissions` | Optional | Specifies the worker's entity permissions. Analogous to [`permissions` in the previous configuration version](https://docs.improbable.io/reference/latest/shared/worker-configuration/permissions), with a difference that the permissions property here is a single permission object rather than an array. |
-| `launchConfiguration` | Required | Describes how the worker can be launched both locally and in the cloud. It contains two fields: <br> <br> - `localDeployment`, which specifies how to start the worker from a pre-built binary file on the local machine. <br> <br>  - `cloudDeployment`, which specifies how to start the worker from the deployment assembly that has been uploaded.
+| `launchConfiguration` | Required | Describes how the worker can be launched both locally and in the cloud. It contains two fields: <br><br> - `localDeployment`, which specifies how to start the worker from a pre-built binary file on the local machine. <br><br>  - `cloudDeployment`, which specifies how to start the worker from the deployment assembly that has been uploaded. |
```
Comment (anne-edwards) [inline docs/reference/server-worker-configuration.md]:
```suggestion
| `launchConfiguration` | Required | Describes how to launch the worker both locally and in the cloud. It contains two fields: <br><br> - `localDeployment`, which specifies how to start the worker from a pre-built binary file on your local machine. <br><br>  - `cloudDeployment`, which specifies how to start the worker from the uploaded deployment assembly. |
```

File: docs/reference/server-worker-configuration.md [hunk 28]
```diff
@@ -48,36 +57,38 @@ There must be a configuration for each platform the worker will be run on.
...
       }
 ```
 
-The platform name can be one of `macos`, `linux` or `windows`. Each of these can have the following fields:
+The platform name can be one of `windows`, `macos` or `linux`. Each of these can have the following fields:
 
 | Field | Required/Optional | Description | 
 | :------------- | :------------- | :------- |
-| `localBuildDirectory` | Required | The directory containing the binary to be executed. |
+| `localBuildDirectory` | Required | The directory containing the binary to be executed. <br><br> **Note:** Your built out worker must be self-contained to the `localBuildDirectory` and all files that your worker might depend on while running must be present in this folder. Files outside of this folder will not be uploaded to the cloud as part of your project assembly. |
```
Comment (anne-edwards) [inline docs/reference/server-worker-configuration.md]:
```suggestion
| `localBuildDirectory` | Required | The directory containing the binary to be executed. <br><br> **Note:** Your built-out worker must be self-contained in the `localBuildDirectory` and all files that your worker might depend on while running must be in this folder. Files outside this folder will not be uploaded to the cloud as part of your project assembly. |
```

File: docs/reference/server-worker-configuration.md [hunk 29]
```diff
@@ -48,36 +57,38 @@ There must be a configuration for each platform the worker will be run on.
...
 | `command` | Required | The base command to be executed to start the worker. |
-| `arguments` | Required | A list of arguments to provide to the base command. The arguments can contain the same placeholder strings as specified in the previous version of configuration, which are interpolated by SpatialOS before the worker is launched. See the [worker launch configuration](https://docs.improbable.io/reference/latest/shared/worker-configuration/launch-configuration) documentation for more information. |
+| `arguments` | Required | A list of arguments to provide to the base command. The arguments can contain the same placeholder strings, like `{IMPROBABLE_RECEPTIONIST_HOST}`, which are interpolated by SpatialOS before the worker is launched. See our documentation on [worker launch configurations](https://docs.improbable.io/reference/latest/shared/worker-configuration/launch-configuration) for more information. |
```
Comment (anne-edwards) [inline docs/reference/server-worker-configuration.md]: It might be my lack of context, but I'm not sure what "the same" refers to here - could you clarify?

File: docs/reference/server-worker-configuration.md [hunk 30]
```diff
@@ -48,36 +57,38 @@ There must be a configuration for each platform the worker will be run on.
...
 
 #### Cloud deployment configuration
-The cloud deployment configuration specifies how to launch the worker in the cloud. It's the same as the local deployment configuration at the moment but will include additional fields in the future, allowing you to specify where the worker binaries are uploaded.
\ No newline at end of file
+The cloud deployment configuration specifies how to launch the worker in the cloud. It's the same as the local deployment configuration at the moment but will include additional fields in the future, allowing you to specify where the worker binaries are uploaded.
+
+Note: It rarely makes sense to specify a cloud deployment configuration for any platform other than `linux`, since workers are run in an Linux environment in the cloud.
```
Comment (anne-edwards) [inline docs/reference/server-worker-configuration.md]: Do we want to tell people when it does make sense to do this, or is it such an edge case that it's not worth going into detail about/people who need to know will already know?

Comment (JonasImprobable) [reply docs/reference/server-worker-configuration.md]: I can personally can't think of a use case where you would want to have non-linux cloud servers.

File: README.md (hunk 12 above)
Comment (JonasImprobable) [reply README.md]: Is `client-worker` (with the `-` in the middle) a convention?

Comment (anne-edwards) [reply README.md]: Yes, server-worker and client-worker are conventions we (tech writers) are trying to push for throughout the docs.

You might also have noticed my adding "type" or "instance" in a couple of places - this is another thing we hope to make consistent, but it's a bit harder because it involves going through every single mention of "worker" and deciding what exactly we're talking about... so for the moment we have to live with some inconsistency.

Comment (JonasImprobable) [reply README.md]: Applied your suggestion.

File: README.md (hunk 13 above)
Comment (JonasImprobable) [reply README.md]: The plan is to have a `migration guide` section containing multiple articles.

File: docs/changelog.md (hunk 14 above)
Comment (JonasImprobable) [reply docs/changelog.md]: That's a great idea. Reversed the order.

File: docs/changelog.md (hunk 15 above)
Comment (JonasImprobable) [reply docs/changelog.md]: Done!

File: docs/changelog.md (hunk 16 above)
Comment (JonasImprobable) [reply docs/changelog.md]: Great idea. Done!

File: docs/migration-guide/faq.md (hunk 17 above)
Comment (JonasImprobable) [reply docs/migration-guide/faq.md]: Nice! Done. :)

File: docs/migration-guide/faq.md (hunk 18 above)
Comment (JonasImprobable) [reply docs/migration-guide/faq.md]: Done.

File: docs/migration-guide/faq.md (hunk 19 above)
Comment (JonasImprobable) [reply docs/migration-guide/faq.md]: Done.

File: docs/migration-guide/faq.md (hunk 20 above)
Comment (JonasImprobable) [reply docs/migration-guide/faq.md]: Done.

File: docs/reference/client-worker-configuration.md (hunk 21 above)
Comment (JonasImprobable) [reply docs/reference/client-worker-configuration.md]: Removed the "between server and client workers" part. Added "instance".

File: docs/reference/client-worker-configuration.md (hunk 22 above)
Comment (JonasImprobable) [reply docs/reference/client-worker-configuration.md]: Nice. Done!

File: docs/reference/client-worker-configuration.md (hunk 23 above)
Comment (JonasImprobable) [reply docs/reference/client-worker-configuration.md]: Done.

File: docs/reference/project-configuration.md (hunk 24 above)
Comment (JonasImprobable) [reply docs/reference/project-configuration.md]: I rephrased the sentence to match your idea.

File: docs/reference/server-worker-configuration.md (hunk 25 above)
Comment (JonasImprobable) [reply docs/reference/server-worker-configuration.md]: Removed.

File: docs/reference/server-worker-configuration.md (hunk 26 above)
Comment (JonasImprobable) [reply docs/reference/server-worker-configuration.md]: Done.

File: docs/reference/server-worker-configuration.md (hunk 27 above)
Comment (JonasImprobable) [reply docs/reference/server-worker-configuration.md]: Done.

File: docs/reference/server-worker-configuration.md (hunk 29 above)
Comment (JonasImprobable) [reply docs/reference/server-worker-configuration.md]: Yeah, this is unclear. I changed the sentence.

File: docs/reference/client-worker-configuration.md [hunk 31]
```diff
@@ -31,19 +31,23 @@ The file may be called by any name but should have the following structure:
...
+    }
   }
 }
 ```
 
 | Field | Required/Optional | Description | 
 | :------------- | :------------- | :------- |
-| `workerType` | Required | Used to identify this worker type elsewhere in your project. |
-|`layer`| Optional | The name of the simulation layer of this worker, used to determine the load balancing strategy. For more information, see [introduction to layers](https://docs.improbable.io/reference/latest/shared/worker-configuration/layers#introduction-to-layers).|
+| `workerType` | Required | Used to identify this worker type elsewhere in your project. <br><br> **Note:** A `workerType` must be unique project-wide. The `workerType` specified in this field must also match the `workerType` that your worker instance reports in its `ConnectionParameters` when connecting to a deployment. For more information, see the [API reference](https://docs.improbable.io/reference/latest/csharpsdk/api-reference#improbable-worker-connectionparameters-class). |
```
Comment (StefaniaChierici) [inline docs/reference/client-worker-configuration.md]: thanks for clarifyng this!
